interp/cfg.go

```diff
@@ -853,28 +854,49 @@ func (interp *Interpreter) cfg(root *node, sc *scope, importPath, pkgName string
 			n.rval = l.rval
 
 		case breakStmt:
-			if len(n.child) > 0 {
-				gotoLabel(n.sym)
-			} else {
+			if len(n.child) == 0 {
 				n.tnext = sc.loop
+				break
+			}
+			if !n.hasAnc(n.sym.node) {
+				err = n.cfgErrorf("invalid break label %s", n.child[0].ident)
+				break
+			}
+			for _, c := range n.sym.from {
+				c.tnext = n.sym.node
 			}
 
 		case continueStmt:
-			if len(n.child) > 0 {
-				gotoLabel(n.sym)
-			} else {
+			if len(n.child) == 0 {
 				n.tnext = sc.loopRestart
+				break
+			}
+			if !n.hasAnc(n.sym.node) {
+				err = n.cfgErrorf("invalid continue label %s", n.child[0].ident)
+				break
+			}
+			for _, c := range n.sym.from {
+				c.tnext = n.sym.node.child[1].lastChild().start
 			}
 
 		case gotoStmt:
-			gotoLabel(n.sym)
+			if n.sym.node == nil {
+				break
+			}
+			for _, c := range n.sym.from {
+				c.tnext = n.sym.node.start
+			}
 
 		case labeledStmt:
 			wireChild(n)
 			if len(n.child) > 1 {
 				n.start = n.child[1].start
 			}
-			gotoLabel(n.sym)
+			for _, c := range n.sym.from {
+				if c.kind == gotoStmt {
+					c.tnext = n.start
```

Comment:
If we are not in a forward case, it seems useless to do that, since the code in the goto case (L887) will overwrite c.tnext.

If we are in the forward case, we are overwriting c.tnext here. Is that what we want to do?